conftest.py

```diff
@@ -0,0 +1,13 @@
+from mypy.test.helpers import PytestSuite
+import inspect
+import pytest
+
+def pytest_addoption(parser):
+    parser.addoption('--update-testcases', action='store_true',
+                     dest='UPDATE_TESTCASES')
+
+def pytest_pycollect_makeitem(collector, name, obj):
+     if (inspect.isclass(obj) and issubclass(obj, PytestSuite) and
+            obj is not PytestSuite):
+        print(name)
+        obj.collect_tests()
```

Comment:
Huh, does this work? The documented interface for the `pytest_pycollect_makeitem` hook which you've implemented is that it should return an appropriate collector:
http://pytest.org/latest/writing_plugins.html#_pytest.hookspec.pytest_pycollect_makeitem

Empirically it seems to work... oh, I see. What's happening is that this implementation has the side effect of setting up a bunch of little methods on the class with names that look like tests; then you return None here, so pytest figures your implementation gave up and falls back to the built-in implementations of this hook, of which the final fallback sees those methods and makes tests of them.

This is a little awkward but it does work. At a minimum it deserves a comment here explaining what's going on, because it isn't using the API provided by pytest in the documented manner.

Alternatively, I think it wouldn't actually be hard to take the core logic you've written and put it into the form of using the documented API. ... I went and read some examples and found myself with a long comment, so I'll put the balance of it in a comment directly on the PR, which feels a little more spacious.